datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -1648,22 +1649,46 @@ fn build_like_match(
             let s = unpack_string(lit.value())?;
             return Ok(s);
         }
-        plan_err!("LIKE expression must be a string literal")
+        plan_err!("Unexpected LIKE expression: {expr:?}")
     }
 
-    // I *think* that ILIKE could be handled by making the min lowercase and max uppercase
-    // but that requires building the physical expressions that call lower() and upper()
+    /// Try and increment the the string's bytes from right to left, returning when the result
+    /// is a valid UTF8 string. Returns `None` when it can't increment any byte.
+    /// Vendored from https://github.com/apache/arrow-rs/blob/56525efbd5f37b89d1b56aa51709cab9f81bc89e/parquet/src/column/writer/mod.rs#L1432-L1448
+    fn increment_utf8(data: &str) -> Result<String> {
+        let mut data = data.as_bytes().to_vec();
+        for idx in (0..data.len()).rev() {
+            let original = data[idx];
+            let (byte, overflow) = original.overflowing_add(1);
```

Comment:
incrementing character 127 should be supported.
same with other utf-8 byte sequences where last byte is saturated.

let's operate on code points, not bytes.

Reply:
Changed to operate on code points in https://github.com/apache/datafusion/pull/12978/commits/5bcca3ef98def407822d23318ac8b6584f9f8c24